week1/homework/src/server.js

```diff
@@ -6,10 +6,33 @@ const http = require('http');
  * will fail.
  */
 function createServer(port) {
+  // TODO: Write your homework code here
   let state = 10;
-
   const server = http.createServer((request, response) => {
-    // TODO: Write your homework code here
+    function responseWrite(statusCode, result = {}) {
+      response.writeHead(statusCode, { 'Content-Type': 'application/json' });
+      response.write(JSON.stringify(result));
+    }
+
+    switch (request.url) {
+      default:
+        responseWrite(404, { error: 'Not found' });
+        break;
+      case '/state':
+        responseWrite(200, { state });
+        break;
+
+      case '/add':
+        responseWrite(200, { state: (state += 1) });
```

Comment:
It's all about readability, this way of reassigning your variable will work just fine, but for a personal taste leaving this `state += 1` on it's own line will make it improvement in that point.
```javascript
state += 1
responseWrite(200, { state });
```

Reply:
thank you, then I will change that as well